### PR TITLE
Update trainer docs

### LINKED ISSUE #295: Trainer: update docs

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #295 and #290

### CODE REVIEW

File: pytorch_pfn_extras/engine.py [hunk 1]
```diff
@@ -116,35 +116,39 @@ def create_trainer(
...
             concluded. The default is an interval trigger set to `max_epochs`
-        writer (writing.Writer, optional):
+        writer:
             Writer that can be used by extensions to write data to custom
             filesystems.
-        evaluator (Evaluator, optional):
+        evaluator:
             Evaluator that is used in evaluation phase.
             If `None` is given, the evaluation is skipped.
+            Evaluators can be created with :func:`pytorch_pfn_extras.engine.create_evaluator`.
```
Comment (kmaehashi) [inline pytorch_pfn_extras/engine.py:139]:
```suggestion
            Evaluators can be created with
            :func:`pytorch_pfn_extras.engine.create_evaluator`.
```

File: docs/source/user_guide/logic.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+a single module.
+
+Logic functions are not exepcted to be directly called by the user.
+They will be invoked by the Trainer and Evaluator engines.
+
+Default Logic (:class:`ppe.handler.Logic <pytorch_pfn_extras.handler.Logic>`)
+------------------------------------------------------------------------------------------
+
+PPE provides a default logic that performs the forward/backward/optimizer loop
+for a single model. This logic allows using some torch features such as AMP
```
Comment (emcastillo) [inline docs/source/user_guide/logic.rst:21]:
```suggestion
for a single model. This logic allows using some torch features such as AMP autocast
```

Sorry, this was probably my mistake in the original docs, AMP is not a feature per see but a package :).

File: docs/source/user_guide/runtimes.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+Use of ``ppe.to`` to transfer modules and batches to custom devices
+---------------------------------------------------------------------------
+
+If you have defined a new runtime for a custom device the ``ppe.to`` function
+allows moving a module or a tensor to the new device by invoking the
+``Runtime.move_tensor`` and ``Runtime.move_module`` when needed.
+
+The module will be tagged by adding a attribute named
+``_ppe_runtime`` that holds the needed runtime. It is the responsibility of the
+user written runtime to perform the actual movement to the device and apply
```
Comment (emcastillo) [inline docs/source/user_guide/runtimes.rst:34]:
```suggestion
user custom runtime to perform the actual movement to the device and apply
```

Maybe this feels better?

File: docs/source/user_guide/trainer.rst [hunk 4]
```diff
@@ -0,0 +1,125 @@
+Trainer and Evaluator
+============================
+
+.. note::
+
+   The Trainer/Evaluator APIs are currently under the technical preview and may subject to change in the future versions.
```
Comment (emcastillo) [inline docs/source/user_guide/trainer.rst:6]:
```suggestion
   The Trainer/Evaluator APIs are currently under technical preview and may subject to change in the future versions.
```

File: docs/source/user_guide/trainer.rst [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+The Trainer and Evaluator provides the device-agnostic training framework for PyTorch.
```
Comment (emcastillo) [inline docs/source/user_guide/trainer.rst:8]:
```suggestion
The Trainer and Evaluator provide the device-agnostic training framework for PyTorch.
```

**Review by emcastillo — CHANGES_REQUESTED** [review]
minor changes

**Review by emcastillo — APPROVED** [review]

**Pull request merged**